lib/arc/file.ex

```diff
@@ -70,12 +74,26 @@ defmodule Arc.File do
     case remote_file do
       {:ok, body} -> File.write(local_path, body)
       {:error, error} -> {:error, error}
+      other -> validate_arc_file(other)
+    end
+  end
+
+  defp validate_arc_file(struct) do
+    cond do
+      struct.__struct__ == Arc.File -> struct
```

Comment:
After we remove our own redirect logic, this `__struct__` and `other` matching should all go away